doc/project/release_process.rst

```diff
@@ -453,10 +453,20 @@ steps:
         #. Post a PR with the updated :zephyr_file:`VERSION` file using
            ``release: Zephyr 1.11.0-rc1`` as the commit subject. Merge
            the PR after successful CI.
+
         #. Tag and push the version, using an annotated tag::
 
             $ git pull
             $ git tag -s -m "Zephyr 1.11.0-rc1" v1.11.0-rc1
+
+        #. Verify that the tag has been signed correctly, ``git show`` for the
+           tag must contain a PGP signature (look for the ``BEGIN PGP
+           SIGNATURE`` marker in the output::
```

Comment:
Do we require a PGP signature or is a SSH signature allowed too? I think we should allow either.

Reply:
Oh, I never realize you could setup the system to sign commits using an ssh key. Reading about it, seems like it still goes through GPG, does the tag output look any different?

Reply:
No, SSH signatures do not go through GnuPG (they go through SSH). The signature looks like this:

```
tag v1.0.0
Tagger: Henrik Brix Andersen <henrik@brixandersen.dk>
Date:   Wed Nov 6 16:34:27 2024

Some Random Project v1.0.0
-----BEGIN SSH SIGNATURE-----
U1NIU0lHAAAAAQAAADMAAAALc3NoLWVkMjU1MTkAAAAgHO3bFNwTb3r/FlLq3yfoSrn12s
vz92G9HJ1lch8JknIAAAADZ2l0AAAAAAAAAAZzaGE1MTIAAABTAAAAC3NzaC1lZDI1NTE5
AAAAQHfuR39Lsw5YbDOyPP0cj2647u87Q4HbRbGPHLYD72TYjZv/5FOOUp/kZd0w9JVoA4
lO65QjjkBSrVtCFK+l1wg=
-----END SSH SIGNATURE-----

...
```

Reply:
Ok, I'm on the edge for this, you need your ssh key to push the tag in the first place so one may argue that signing with the ssh key is redundant and we should ask for a gpg one. Or we may be happy with both, what do you think? I'd be glad to dump that key myself to be perfectly honest, I only use it for signing tags.

Reply:
You use a dedicated SSH key for signing (at least in GitHub). When adding SSH key in the GitHub UI, you need to specify whether the SSH key is to be used for signing or authentication.

My point is that Git (and GitHub) supports both GPG/PGP and SSH keys for signing, not sure why we as a project should restrict ourselves to one of these types.

Reply:
ok so just to be clear, you can't use the same key for pushing and signing? then I think there's no argument, either are fine (also now that you point it out I remember reading about it and figuring I'd keep the gpg one, can't recall why, c'mon brain do it!)

Reply:
Either way, what are we actually trying to accomplish, btw, with these signed commits? I mean, I understand the value of signing, of course, but in this case what's the value of using what's effectively a self-signed GPG key / SSH certificate to sign? It looks like most of the public keys of the recent past release managers are not at all part of a web of trust since they're not signed by anyone else but their owner, ... who could be anyone, right? I think for the GPG signature to really have impact one would need to be able to trace it back to at least a few other people from the community, and ideally to folks they *directly* trust. Signing party at the next ZDS anyone? :) 

Of course one can check that the commit signature matches what they can find on https://github.com/{RELEASE_MANAGER}.keys or https://github.com/{RELEASE_MANAGER}.gpg, and AFAICT the zephyrproject-rtos requires 2FA, so there's that, but I am not sure that's enough? IMO the signatures should have "value" independently of whatever additional info can be gathered from GitHub at a given point in time.

Reply:
(I was typing my response during the TSC meeting and it was meant to be a reply/follow up to https://github.com/zephyrproject-rtos/zephyr/pull/80962#discussion_r1831380388. I see that @henrikbrixandersen  made more or less the same points in the meantime.

Reply:
> ~Perhaps we need to take a step back and try to determine _why_ we require tags to be signed in the first place:~
> 
> * ~Is to get the "Verified" badge next to our GitHub releases?~
> * ~Is it for downstream users to be able to verify the authenticity of the tag?~
> * ~Is it for downstream users to be able to verify the identify of the person making the tag?~

@henrikbrixandersen I think this is still very much an open topic :) And for bullet 2 and 3, if we agree these are part of the goals, the verification step is something that users should be able to do with zero dependency on GitHub whatsoever

Reply:
> Either way, what are we actually trying to accomplish, btw, with these signed commits?

how useful they are and what they serve is now irrelevant imo, we started signing some time ago and if we stop signing it will look odd, i.e. look at https://github.com/zephyrproject-rtos/zephyr/tags, all verified and signed tags, if you release something that is not signed now because we think it is useless, that will immediately pop out. So, let's just keep doing that and let us consider how we can establish the web of the trust going forward, so it is more effective.

Reply:
> how useful they are and what they serve is now irrelevant imo, we started signing some time ago and if we stop signing it will look odd,

@nashif Mmm really? Has anyone ever complained about 3.5.0 then? Cause it sure pops out :) https://github.com/zephyrproject-rtos/zephyr/tags?after=v3.6.0-rc2

And to be clear I (nor anyone else I think) never said we should stop signing or signing releases is not useful

Reply:
> And to be clear I (nor anyone else I think) never said we should stop signing or signing releases is not useful

Agreed. I think we need to establish if signing should allow for root-of-trust validation (which I believe requires signing with GPG keys signed by other community members along with some sort of key infrastructure as outlined by @kartben above), or if we "just" want verified releases/tags (which can be done with either key type).

Reply:
> Of course one can check that the commit signature matches what they can find on https://github.com/{RELEASE_MANAGER}.keys 

Please note that the keys listed there are authentication keys, not signing keys.

Reply:
> > Of course one can check that the commit signature matches what they can find on [@{RELEASE_MANAGER}.keys](https://github.com/%7BRELEASE_MANAGER%7D.keys)
> 
> Please note that the keys listed there are authentication keys, not signing keys.

Ah yes, good point! https://api.github.com/users/RELEASE_MANAGER_USER_NAME/ssh_signing_keys

Reply:
> @nashif Mmm really? Has anyone ever complained about 3.5.0 then? Cause it sure pops out :) [zephyrproject-rtos/zephyr/tags?after=v3.6.0-rc2](https://github.com/zephyrproject-rtos/zephyr/tags?after=v3.6.0-rc2)

the silent majority will often just move on and and make its own conclusions about the project and inconsistencies it has, you do not need to get an official complaint. (btw, someone mentioned yesterday the issue was noticed but users and they though something fishy was going on).

Anyways. you asked the question of usefullness, I did not take this as  you wanting them to be dropped, was just answering.

Reply:
> Of course one can check that the commit signature matches what they can find on [https://github.com/{RELEASE_MANAGER}.keys](https://github.com/%7BRELEASE_MANAGER%7D.keys) or [https://github.com/{RELEASE_MANAGER}.gpg](https://github.com/%7BRELEASE_MANAGER%7D.gpg), and AFAICT the zephyrproject-rtos requires 2FA, so there's that, but I am not sure that's enough?

That was somewhat how I interpreted it. Anyway I changed it to say PGP or SSH, I guess for the purpose of making things look green on GH either works so whatever.

Would be open to expand the conversation though, cross signing keys for release team members does not sound like a bad idea.

Reply:
> but in this case what's the value of using what's effectively a self-signed GPG key / SSH certificate to sign?

I've had this question often, especially when having to remake a GPG key to sign a release here.  Likely need to establish a secure vault and GitHub automation to do the signing through a key verified/trusted by the TSC (or maybe Release WG) members.  The PR in question isn't going to run down any of this.